### DISCUSSION TITLE
When making a request with HTTP cache set, response has status code 304 – Will response still include cached data?

### DISCUSSION THREAD

**Discussion by iandees** [sindresorhus/got#1778] [Q&A] [answered]
I'm running into a case in my production system that I'm having a hard time reproducing, so I thought I'd ask what the system was designed to do.

I'm using `got` to make a request with the `cache` option set to a QuickLRU instance. The status code of the response is sometimes an HTTP 304. I'm not adding any special headers to the request, so I'm assuming that `got`'s cache is adding them for me and the server is (correctly) responding with a 304 to tell the client the data in cache hasn't changed. Is that assumption right?

If it is, can I safely assume that the response object's `body` property will include the cached data from the QuickLRU if the response code is 304?

**Comment by szmarczak** [comment, marked as answer]
Correct.

**Reply by iandees** [reply]
Great, thank you!
